crates/slice-dst/examples/my_vec.rs

```diff
@@ -0,0 +1,126 @@
+use std::{
+    fmt::Display,
+    mem::{self, MaybeUninit},
+    ops::Deref,
+};
+
+use ref_cast::RefCast;
+use slice_dst::SliceWithHeader;
+
+/// Default capacity of [`MyVec`].
+const MY_VEC_DEFAULT_CAPACITY: usize = 4;
+
+/// On the heap we will store the number of used elements in the slice (length)
+/// and a slice of (maybe uninitialized) values.
+type HeapData<T> = SliceWithHeader<usize, MaybeUninit<T>>;
+
+/// Our [`Vec`] implementation.
+///
+/// _Note:_ In contrast to [`std::vec::Vec`] this stores its length on the heap.
```

Comment:
suggestion: use [`erasable::Thin`](https://cad97.github.io/pointer-utils/erasable/struct.Thin.html) and store `Thin<Box<HeapData<T>>>`, so the pointer is thin (single-`usize`-big) and length/capacity are both exclusively on the heap. `SliceWithHeader` stores its (slice) length inline to support erasing. (The eventual custom derive will omit this.)

Reply:
I'm a little bit reluctant to introduce `erasable::Thin` here. It addresses another issue and is from another crate. I would prefer to stick to an honorable mention here and put another example in the `erasable` crate.